Feat. MyRecordView 구현 (#15)

RoastedJM opened the pull request:
Why: 사용자가 MyPage에서 본인이 남긴 전체 기록을 확인할 수 있도록 하기 위함

How:
- SwiftData 기반 목 데이터 모델링
- Card, List의 컴포넌트화

<!--
🙏 PR 제목 컨벤션 (Gitmoji + 타입 + 이슈 번호 + 작업 요약)
예시: ✨ Feat: #167 예약 취소 구현
※ PR 생성 시 Assignees 및 Labels 설정도 잊지 마세요!
-->

## ✨ What’s this PR?
### 📌 관련 이슈 (Related Issue)
<!-- 해당 PR이 어떤 이슈를 해결하는지 연결해주세요 -->
- Closes #8 

---

### 🧶 주요 변경 내용 (Summary)
<!-- 이번 PR에서 작업한 핵심 변경 사항을 작성해주세요 -->

- 사전에 논의된 ERD를 기반으로 하는 목 데이터 모델링
- Card를 만드는 부분과 List를 만드는 부분 컴포넌트화

---

### 📸 스크린샷 (Optional)
<!-- UI 작업의 경우, 구현한 화면을 첨부해주세요 -->
<!-- 이미지 크기 조절 예시: <img width="300" alt="설명" src="링크"> -->
<img width="359" height="695" alt="스크린샷 2025-07-20 오후 2 02 08" src="https://github.com/user-attachments/assets/6f0f2038-535b-4fe6-ab8c-5a09f86077a7" />


---

### 🧪 테스트 / 검증 내역
<!-- 동작 확인 여부나 시나리오 테스트 내용을 간단히 써주세요 -->

- [x] UI 정상 동작 확인

---

### 💬 기타 공유 사항
<!-- 리뷰어가 참고하면 좋을 정보, 고민했던 지점 등을 적어주세요 -->

- Asset에 이미지를 넣어도 불러올 수 없어, 확인해보니 패키지와 연관된 부분인 것 같아 추가 작업이 필요할 것 같습니다.

---

### 🙇🏻‍♀️ 리뷰 가이드 (선택)
<!-- 리뷰어가 중점적으로 보면 좋을 포인트가 있다면 알려주세요 -->

- 데이터 구조 
- flower enum의 사용 방식
- 기본적인 사항들

RoastedJM committed:
Feat. MyRecordView 구현 (#15)

Why: 사용자가 MyPage에서 본인이 남긴 전체 기록을 확인할 수 있도록 하기 위함

How:
- SwiftData 기반 목 데이터 모델링
- Card, List의 컴포넌트화

RoastedJM commented on Packages/Features/Sources/Features/MyRecordView/MyRecordModel.swift:
```diff
@@ -0,0 +1,129 @@
+        ),
+        ObjectEntity(
+            userId: mockUser.id,
+            title: "C5 세션이 끝난 뒤",
+            createdAt: Calendar.current.date(from: DateComponents(year: 2024, month: 3, day: 7))!,
+            latitude: 36.4151,
+            longitude: 129.3665,
+            address: "경북 포항시 남구 청암로 77",
+            flower: .sunflower
+        ),
+        ObjectEntity(
+            userId: mockUser.id,
+            title: "C5 세션이 끝난 뒤",
+            createdAt: Calendar.current.date(from: DateComponents(year: 2024, month: 3, day: 7))!,
+            latitude: 36.4151,
+            longitude: 129.3665,
+            address: "경북 포항시 남구 청암로 77",
```
목업 데이터로 임시로 넣어둔 부분이기에 상관 없음

freeskyES reviewed (commented):
와우!! 고생하셨습니다~ 
`MyRecordModel` 파일에 데이터와 enum, 컴포넌트 등 잘 정리해주셔서 보기 좋았습니다.

다만 한 가지 제안드릴 점은, 현재 파일 안에 SwiftData의 `@Model`이 함께 포함되어 있는데요,

- MyRecordModel 에서 쓰이는 SwiftData 용 @Model 를 제거해주시면 좋을 것 같아요!
(`Features` 모듈에서는 SwiftData 등 저장소 관련 의존성은 분리해두는 게 좋아요 )
- 대신 **MyRecordView에서 필요한 데이터만 정리한 모델**로 구성해보시면 어떨까요?

---
정리해보면,
1. 저희가 다같이 정의한 **공통 엔티티(ObjectEntity 등)** 대신  
→ View 전용으로 간단하게 정의한 `MyRecordModel`을 새로 만들어주세요 (MyRecordView에 필요한 데이터만)

2. ViewModel을 하나 만들어서 
→ 데이터를 가공 후 바인딩하면 MVVM 구조에 더 적합하고 유지보수가 쉬워질 것 같습니다!

(아래는 예시로 참고만 해주세요!)

```swift
struct MyRecordModel: Identifiable {
    let id: UUID
    let title: String
    let formattedDate: String
    let flowerImageName: String
}
```

MyRecordViewModel 예시 포멧
```swift
final class MyRecordViewModel: ObservableObject {
    @Published var records: [MyRecordModel] = []

    init() {
        loadMockData()
    }

    private func loadMockData() {
        records = MyRecordModel.createMock()  // 가베가 만들어두신 MockDataManager -> MyRecordModel 안/Extension/Mapper 클래스  등등 많은 방법으로 createMock() 함수 작성할 수 있을 것 같습니다.
    }
}

RoastedJM committed:
Refactor. SwiftData 의존성 제거 및 패턴 변경

Why: 더 나은 협업을 위한 SwiftData 의존성 제거와 MVVM패턴으로의 변경
How:
- SwiftData @Model 클래스들(ObjectEntity, User) 제거
- View 전용 MyRecordModel struct 생성
- MyRecordViewModel을 ObservableObject로 구현하여 데이터 바인딩
- MockDataManager 대신 MyRecordModel.createMock() static 메서드로 대체
- 날짜 포맷팅 로직을 ViewModel에서 처리하도록 변경
- RecordCard, RecordList에서 모델 타입을 MyRecordModel로 변경

RoastedJM committed:
Chore. Merger origin into dev

Why: 작업중이던 내용 개인 브랜치로 병합
How: git merge develop

RoastedJM committed:
Chore. Merge Develop into feature

Why: dev 내용 머지
How: 깃 머지

RoastedJM committed:
Feat. Added Image Asset

Why: Design System 추가에 따른 변경
How: 추가된 Design System Import 후 이미지 추가

freeskyES reviewed (approved):
가베~ 짱인데요! 반영해주셔서 감사합니다.
